app/src/main/java/org/schabi/newpipe/player/PlayerService.java

```diff
@@ -177,9 +299,55 @@ public static class LocalBinder extends Binder {
         public PlayerService getService() {
             return playerService.get();
         }
+    }
+
+    /**
+     * @return the current active player instance. May be null, since the player service can outlive
+     * the player e.g. to respond to Android Auto media browser queries.
+     */
+    @Nullable
+    public Player getPlayer() {
+        return player;
+    }
 
-        public Player getPlayer() {
-            return playerService.get().player;
+    /**
+     * Sets the listener that will be called when the player is started or stopped. If a
+     * {@code null} listener is passed, then the current listener will be unset. The parameter taken
+     * by the {@link Consumer} can be null to indicate that the player is stopping.
+     * @param listener the listener to set or unset
+     */
+    public void setPlayerListener(@Nullable final Consumer<Player> listener) {
+        this.onPlayerStartedOrStopped = listener;
+        if (listener != null) {
+            if (player == null) {
+                listener.accept(null);
+            } else {
+                listener.accept(player);
+            }
         }
```

Comment:
Hi, I'm leaving a small suggestion here:
```suggestion
    if (listener == null) {
        return;
    }
    listener.accept(player);
```

Reply:
I think the reason it's written like this is that I wanted to make it explicit that the listener might be called with a null value, but probably a short comment would explain it better. Thanks for the catch ;-)

Reply:
```suggestion
        if (listener != null) {
            // if there is no player, then `null` will be sent here, to make sure the state is synced
            listener.accept(player);
        }
```